### ISSUE TITLE
How to overwrite generated swagger.json?

### ISSUE THREAD

**Issue by jtorres-viavansi** [swagger-api/swagger-core#1878]
Good day, I'm having a trouble with generated swagger.json as I described in [this](https://github.com/swagger-api/swagger-ui/issues/2322) other issue. Since it seems to be implemented on future 3.0 spec, I'm thinking in a temporal workaround: overwirte manually generated swagger.json. 

Actually with MULTIPART-FORM-DATA the types are not generated correctly, so my generated swagger.json for one bean in my method is the following:

          "name" : "verificationRequestDTO",
          "in" : "formData",
          "description" : "VerificationRequestDTO object",
          "required" : true,
          "type" : "ref"

I'd like to replace for its correct type as follows:

          "name" : "verificationRequestDTO",
          "in" : "formData",
          "description" : "VerificationRequestDTO object",
          "required" : true,
          "schema" : {
            "$ref" : "#/definitions/VerificationRequestDTO"
          }

Is there any way to make this? Thanks for your help.

**Comment by fehguy** [comment]
You should post your annotations here for completeness.

I believe you can mark your `verificationRequestDTO` as hidden with `hidden = true` and add an `@ApiImplicitParam`, which will let you "lie" about the data type.  Just set it as a string.

**Comment by jtorres-viavansi** [comment]
Sorry, here goes the annotations:

    @POST
    @Path("/document")
    @Consumes(MediaType.MULTIPART_FORM_DATA)
    @Produces(MediaType.APPLICATION_JSON)
    @ApiOperation(value = "Verifies the signature in the passed document",
        notes = "Verifies the signature in the passed document",
        response = VerifierDocumentResponseDTO.class)
    @ApiResponses(value = {
        @ApiResponse(code = 200, message = "ResponseVerifierDocumentDTO with verification result", response = VerifierDocumentResponseDTO.class),
        @ApiResponse(code = 401, message = "Unauthorized", response = VerifierDocumentResponseDTO.class),
        @ApiResponse(code = 403, message = "Forbidden", response = VerifierDocumentResponseDTO.class),
        @ApiResponse(code = 500, message = "Internal server error", response = VerifierDocumentResponseDTO.class)
    })
    public VerifierDocumentResponseDTO verifyDocument(
            @ApiParam(value = "Authentication string", required = true) @HeaderParam("Authorization") String authString,
            @ApiParam(value = "VerificationRequestDTO object", required = true) @FormDataParam("verificationRequestDTO") VerificationRequestDTO verificationRequestDTO, 
            @ApiParam(value = "Original document to verify", required = true) @FormDataParam("originalDocument") InputStream originalDocument, 
            @ApiParam(value = "Signed document", required = true) @FormDataParam("signedDocument") InputStream signedDocument) {

Thanks.

**Comment by webron** [comment]
Not really sure what you're trying to achieve. 3.0 is not finalized yet, there are no tools for it, and it contains many other structural changes. Are you planning on hacking your way around all the relevant tools in the chain? You can always write a filter that will manipulate the spec however you want but that alone will not help you much.

**Comment by jtorres-viavansi** [comment]
Hi @webron , I thought in that solution, a filter to replace only that strings since the rest of the spec is correctly generated.

Thanks for your help.

**Comment by webron** [comment]
But even if you replace it, it won't help. No tool will be able to understand it, so what are you hoping to gain?

**Comment by jtorres-viavansi** [comment]
Replacing it works, I've tried it. I've created swagger_fixed.json replacing manually the original json for the fixed one and I get good results:

1. swagger-ui works fine with fixed json and methods are correctly invoked. In fact, the beans in MULTIPART-FORM-METHODs are correctly presented.
2. swagger-codegen generates fine the code. With not fixed one I get errors generating objc model for example.

The fixed swagger.json is supossely a valid one, isn't it? Thanks.
